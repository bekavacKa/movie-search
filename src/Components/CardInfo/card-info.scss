@import '../../Assets/scss/variables.scss';
@import '../../Assets/scss/mixins.scss';

.card-info{
    @include my-flex(center, column, center);
    background-color: $secondBackColor;
    box-shadow: 0 0 7px 1px $firstFontColor;
    width: 280px;
    padding: 0;
    border-radius: 5px;
    transition: $transition;
    color: $firstBackColor;
    cursor: pointer;
    &:hover{
        box-shadow: 0 0 7px 2px $firstFontColor;
        transform: scale(1.01);
    }
    .card-info-image{
        width: 100%;
        @include my-flex(center, column, center);
        img{
            width: 100%;
            border-radius: 5px 5px 0 0;
        }
    }
    .card-info-more{
        @include my-flex(center, column, center);
        padding: 35px 5px 25px;
        width: 100%;
        flex: 1;
        position: relative;
        background: linear-gradient(rgba(239, 215, 187, 1), rgba(239, 215, 187, 1), rgba(239, 215, 187, .3));
        .card-info-more-average{
            @include my-flex(center, column, center);
            position: absolute;
            background: linear-gradient(rgba(239, 215, 187, .5), rgba(239, 215, 187, 1), rgba(239, 215, 187, 1));
            box-shadow: 0 0 3px 2px $firstFontColor;
            top: -30px;
            height: 60px;
            width: 60px;
            border-radius: 50%;
            font-weight: 900;
            text-shadow: 1px 1px 1px black;
            font-size: 1.5rem;
            letter-spacing: .1rem;
            &.bad-av{
                color: red;
            }
            &.good-av{
                color: orange;
            }
            &.super-av{
                color: rgb(4, 168, 4);
            }
        }
        .card-info-more-title{
            text-transform: uppercase;
            text-align: center;
        }
    }
}

// @media only screen and (min-width: 320px) and (max-width: 768px){
// }