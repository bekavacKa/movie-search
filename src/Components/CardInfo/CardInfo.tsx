import React from 'react';
import './card-info.scss';

function CardInfo() {


  return (
    <div className='card-info'>
        <div className='card-info-image'>
            <img src='https://mir-s3-cdn-cf.behance.net/project_modules/max_1200/c618cd88432989.5dd5e72e505d1.jpg' alt='bla' />
        </div>
        <div className='card-info-title'>
            <h2>seven</h2>        
        </div>
    </div>
  )
}

export default CardInfo