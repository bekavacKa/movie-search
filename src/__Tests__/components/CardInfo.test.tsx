import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CardInfo from '../../Components/CardInfo/CardInfo';

test('Snapshot test', () => {
    const initialState = { 
        searchTermStore: '',
        buttonsStore:{
            tvShowsButton: true,
            movieButton: false
        },
        storeSearchTerm: ""
    };
    const mockStore = configureStore();
    let store;

    store = mockStore(initialState);
  const tree = renderer.create(
    <Provider store={store}>
        <Router>
            <CardInfo />
        </Router>
    </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
