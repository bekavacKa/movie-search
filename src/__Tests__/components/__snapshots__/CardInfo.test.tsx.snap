// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<a
  className="card-info"
  href="/details/undefined"
  onClick={[Function]}
>
  <div
    className="card-info-image"
  >
    <img
      src="https://static8.depositphotos.com/1009634/988/v/600/depositphotos_9883921-stock-illustration-no-user-profile-picture.jpg"
    />
  </div>
  <div
    className="card-info-more"
  >
    <h2
      className="card-info-more-title"
    />
  </div>
</a>
`;
